### PR TITLE
mention `Cmd(::Vector{String})` in `Cmd` docstring

### PR DESCRIPTION
**Pull request by sjkelly**
This seems like a cyclical definition without it, and
naively `Cmd` and `String` seem on the surface to be
similar so it can be surprising that `Cmd("a string")`
does not work. Adding this signature in the docstring
should obviate the issues for first time users that read
the docstring without having reviewed the manual.

### CODE REVIEW

File: base/cmd.jl [hunk 1]
```diff
@@ -41,6 +41,7 @@ has_nondefault_cmd_flags(c::Cmd) =
 
 """
     Cmd(cmd::Cmd; ignorestatus, detach, windows_verbatim, windows_hide, env, dir)
+    Cmd(exec::Vector{String})
```
Comment (stevengj) [inline base/cmd.jl:44]: Should be noted that this constructor does *not* copy `exec`, so any subsequent changes to `exec` will be reflected in the `Cmd` object.

**Comment by sjkelly** [comment]
Thanks for the feedback @stevengj !

File: base/cmd.jl [hunk 2]
```diff
@@ -70,8 +71,15 @@ while changing the settings of the optional keyword arguments:
 * `dir::AbstractString`: Specify a working directory for the command (instead
   of the current directory).
 
-For any keywords that are not specified, the current settings from `cmd` are used. Normally,
-to create a `Cmd` object in the first place, one uses backticks, e.g.
+For any keywords that are not specified, the current settings from `cmd` are used.
+
+Note that the `Cmd` constructor does not create a copy of `exec`. Any subsequent changes to `exec` will be reflected in the `Cmd` object.
+
+The most common way to construct a `Cmd` object is with command literals, e.g.
```
Comment (stevengj) [inline base/cmd.jl:78]:
```suggestion
The most common way to construct a `Cmd` object is with command literals (backticks), e.g.
```

File: base/cmd.jl [hunk 3]
```diff
@@ -70,8 +71,15 @@ while changing the settings of the optional keyword arguments:
 * `dir::AbstractString`: Specify a working directory for the command (instead
   of the current directory).
 
-For any keywords that are not specified, the current settings from `cmd` are used. Normally,
-to create a `Cmd` object in the first place, one uses backticks, e.g.
+For any keywords that are not specified, the current settings from `cmd` are used.
+
+Note that the `Cmd` constructor does not create a copy of `exec`. Any subsequent changes to `exec` will be reflected in the `Cmd` object.
```
Comment (stevengj) [inline base/cmd.jl:76]:
```suggestion
Note that the `Cmd(exec)` constructor does not create a copy of `exec`. Any subsequent changes to `exec` will be reflected in the `Cmd` object.
```

**Review by stevengj — APPROVED** [review]

**Pull request merged**